library-best-practices/JLBP-6.md

```diff
@@ -0,0 +1,141 @@
+[JLBP-6] Avoid creating conflicts between major versions
+------------------------------------------------------------------------
+
+- A new major version should generally use a different Java package.
+- There is a tradeoff: if an extremely large library (for example thousands of
+  classes) breaks a single method, the cost of a package rename may be higher
+  than dealing with the fallout of dealing with diamond dependency conflicts for
+  that particular method. Only perform an in-place breakage if users can easily
+  replace their usage of the old library with the new library. The larger the
+  portion of a library that breaks between major versions, the higher the value
+  is from renaming the java package.
+- *If* the Java package is changed, use different maven coordinates (for example
+  different artifact id).
+- *If* the Java package is kept the same, use the same maven artifact ID.
+- In other words, rename the Java package if and only if the maven coordinates
+  change.
```

Comment:
Need to specify what is a _coordinates change_ as even bumping the micro version is a coordinate change.  Which contradicts the first point (line 4).

Reply:
Ahh, good point about the meaning of "coordinates" for Maven - I wasn't being careful there. I think many of my uses of the word "coordinates" in this doc are misleading, so I'll fix that.